### ISSUE TITLE
Bug: sdkman detects more recent Java version as older

### ISSUE THREAD

**Issue by iemejia** [sdkman/sdkman-cli#898]
**Bug report**

When I run `sdk upgrade` and I have multiple Java versions installed if one of those is older than the current 'latest' sdkman suggest me to upgrade to it, even if it is an 'older' version.

```bash
sdk upgrade

Upgrade:
java (16.0.0.hs-adpt, 8.0.282.hs-adpt, 11.0.11.hs-adpt, 21.1.0.r11-grl < 11.0.10.hs-adpt)

Upgrade candidate(s) and set latest version(s) as default? (Y/n): n
```

**To reproduce**

Install the Java versions above and run `sdk upgrade`

**System info**

Linux Ubuntu 20.04.2
 zsh 5.8 (x86_64-ubuntu-linux-gnu)
SDKMAN 5.11.0+644

**Comment by hisnameishb** [comment]
I have the same bug:
sepp@box:~/Schreibtisch$ sdk ug

Upgrade:
java (11.0.11-open, 16.0.1-open, 16.0.1-librca, 11.0.10.9-ms, 11.0.11.hs-adpt, 6.0.119-zulu, 17.ea.18-open, 8.0.292-zulu, 21.1.0.r16-grl, 11.0.11-librca, 8.0.292.hs-adpt, 8.0.292-open, 7.0.302-zulu < 11.0.10.hs-adpt)

Ubuntu 20.10, bash, SDKMAN 5.11.0+644

**Comment by nicokosi** [comment]
Same issue on:

macOS Big Sur (11.3.1)
zsh 5.8 (x86_64-apple-darwin19.6.0)
SDKMAN 5.11.2+698

**Comment by marc0der** [comment]
This is not a bug. SDKMAN has a default version for every candidate. The default version for Java is set to the current LTS version, which happens to be the latest in the series of 11 right now.

The name `upgrade` was not very well chosen, as it implies that the latest should be installed. I'm open to suggestions for a better name for this command.

**Comment by nicokosi** [comment]
> The default version for Java is set to the current LTS version, which happens to be the latest in the series of 11 right now.

In my case (it should maybe not appear in this issue :innocent:), I don't understand why the default version `11.0.10.hs-adpt` is older than my installed version, `11.0.11.hs-adpt`:

> 11.0.11.hs-adpt, 16.0.1.hs-adpt, 8.0.292.hs-adpt; default: 11.0.10.hs-adpt

**Comment by iemejia** [comment]
Exact, it seems that there are two issues not detecting the full version correctly mentioned by @nicokosi and the confusing name, but maybe it will be my clear if we don't use the < sign because Java 16 is not < than Java 11:

java (16.0.0.hs-adpt, 8.0.282.hs-adpt, 11.0.11.hs-adpt, 21.1.0.r11-grl **<** 11.0.10.hs-adpt)

**Comment by marc0der** [comment]
Yeah, that was changed long ago. There is no longer a <. Perhaps you are on an old version @iemejia.

**Comment by iemejia** [comment]
You are right @marc0der now it looks better, it seems however we have not set up the latest default version correctly to be 11.0.11.hs-adpt

```
$ sdk upgrade

Available defaults:
groovy (local: 2.5.14; default: 3.0.8)
java (local: 8.0.292.hs-adpt, 17.ea.20-open, 11.0.11.hs-adpt, 21.1.0.r11-grl, 16.0.1.hs-adpt; default: 11.0.10.hs-adpt)
```

**Comment by marc0der** [comment]
Glad to hear it! I think @eddumelendez might be able to help with this as he curates the Java candidate versions.
